### ISSUE TITLE
Entering and leaving directories

### ISSUE THREAD

**Issue by shanesturrock** [swcarpentry/shell-novice#1108]
It looks like there has been discussion on the orientation of directory trees (I'm old so for me the root is at the bottom but I get that others prefer to consider it like a family tree) but I wonder if we could avoid enforcing the orientation by the use of up and down and rather talk about entering and exiting directories? That's what I do when teaching people to navigate the file system, treating directories like rooms that you enter and leave. That way you can describe `cd ..` as leaving the current directory and `.` as staying in the current directory which I think is more helpful to learners. It also helps when describing relative paths I find.

**Comment by gcapes** [comment]
Hi @shanesturrock 
I don't remember this. Could you point me to the discussion please?

When teaching you could use this analogy, but directories (except root) are contained within other directories, so the analogy immediately breaks down because rooms are usually not contained within another room.

I would have thought that most learners will be familiar with a file browser like Windows Explorer so the idea of levels won't be new, and when you expand a subdirectory in a file browser, the contents show below. When I've used Finder on mac, the same concepts apply but it's all rotated 90 degrees because mac.

TL;DR
I think up/down are standard conventions for talking about this. Or aren't they?

**Comment by shanesturrock** [comment]
I hunted through the previous issues where I found this previous [discussion](https://github.com/swcarpentry/shell-novice/issues/450#issue-178338036) which is basically bringing up the top down or bottom up issue and I note in the material as it stands there's discussion around the root being at the bottom. 30 years ago I was taught the root was at the base of the tree and you move up into branches as that was the convention but with the appearance of graphical file managers the visual layout has somewhat taken over. I think the problem is we do talk about going into a directory as a common convention but then we talk about going down (or up) to leave which seems at odds. I try and be consistent by using the entering or opening a directory and exiting or closing.

As a Mac user (who primarily uses Terminal.app because Finder is so clumsy) there are many ways to display the directory tree and it's rarely up or down. The standard view in Finder is icons and when you double click you'll go into that folder and you get back out by hitting the back button. Each icon is like a door to a room, and that room can have multiple doors to other rooms. This is getting somewhat convoluted and you're probably right that most people understand a file manager although I've encountered many who don't even though they use them daily and I've had success in teaching people how to visualise where they are by the room analogy so thought it was worth throwing out there.

**Comment by gcapes** [comment]
Yes, Finder is a bit of a special case and doesn't lend itself to visualizing where you are in the file system. So this puts novice mac users at a bit of a disadvantage.

The lesson teaches the concept of a hierarchical file system, and in doing so we should follow conventions in terms of the terminology used. This will enable learners to communicate effectively with others.

As such I propose that the lessons cover this terminology explicitly rather than assuming knowledge (as it currently does).
i.e. what is meant by (down/up a level, and into/out of a directory), and how file browsers like Windows Explorer might have influenced this terminology. 

e.g. Windows Explorer (and similar) have an 'up' button to go up a level, which is kinda the opposite of double clicking to 'go into' a directory.

https://www.howtogeek.com/wp-content/uploads/2016/06/00a_file_explorer_windows_10_style.png

For me, the rooms analogy breaks down too quickly, so I think this would be best left for face to face explanations.

**Comment by shanesturrock** [comment]
Fair enough, at least this discussion will come up when someone else has the same idea and the course already talks about places so there's room to try multiple ways to explain it to users.

**Comment by gcapes** [comment]
Does that change help to address the issue?

**Comment by shanesturrock** [comment]
Yep, that works thanks.
